wicketd/src/transceivers.rs

```diff
@@ -0,0 +1,372 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Fetching transceiver state from the SP.
+
+use gateway_client::types::{SpIdentifier, SpType};
+use omicron_common::api::external::SwitchLocation;
+use slog::{debug, error, Logger};
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+    time::Duration,
+};
+use tokio::{
+    sync::{mpsc, watch},
+    time::Instant,
+};
+use transceiver_controller::{message::ExtendedStatus, SpRequest};
+use transceiver_controller::{ConfigBuilder, Controller, Error, ModuleId};
+use wicket_common::inventory::Transceiver;
+
+/// Type alias for a map of all transceivers on each switch.
+pub type TransceiverMap = HashMap<SwitchLocation, Vec<Transceiver>>;
+
+// Queue size for passing messages between transceiver fetch task.
+const CHANNEL_CAPACITY: usize = 4;
+
+// Duration on which we poll transceivers ourselves, independent of any
+// requests.
+const TRANSCEIVER_POLL_INTERVAL: Duration = Duration::from_secs(5);
+
+// IP interface we use when polling transceivers on the local switch.
+const LOCAL_SWITCH_SP_INTERFACE: &str = "sidecar0";
+
+// IP interface we use when polling transceivers on the other switch.
+const OTHER_SWITCH_SP_INTERFACE: &str = "sidecar1";
```

Comment:
This is the same no matter whether we're on sled 14/switch 0 or 16/switch 1? Might be worth an explanation in that case.

edit: ahh, saw the comment below. Would be worth saying something like "irrespective of whether this is switch 0 or switch 1".

Reply:
Yes, that's right. `sidecar0` always means "The SP on the switch I'm on" and `sidecar1` means "The other switch's SP". I don't love that these are consts, but I couldn't see a good argument for making them configurable. Unless we address https://github.com/oxidecomputer/dendrite/issues/221....Actually, I'm going to add a reference to that issue here for myself :)